src/charm.py

```diff
@@ -291,6 +201,40 @@ def _common_exit_hook(self) -> None:  # noqa: C901
                 logger.info("CONFIGURED state reached")
             self.unit.status = ActiveStatus()
 
+    def _on_sync_now_action(self, event: ActionEvent):
+        """Hook for the sync-now action."""
+        if not self.container.can_connect():
+            event.fail("Container not ready")
+            return
+
+        event.log("Calling git-sync with --one-time...")
+
+        try:
+            process = self.container.exec(self._command(one_time=True))
+        except APIError as e:
+            event.fail(str(e))
+            return
+
+        try:
+            stdout, warnings = process.wait_output()
+        except ExecError as e:
+            for line in e.stderr.splitlines():
+                event.log(line)
+            event.fail("Exited with code {e.exit_code}.")
+            return
+        except ChangeError as e:
+            event.fail(str(e))
+            return
```

Comment:
Extract try/catch pairs into separate (private) methods.

Reply:
What do you mean?